qfieldsync/gui/cloud_converter_dialog.py

```diff
@@ -0,0 +1,247 @@
+            self.network_manager.projects_cache.refresh()
+
+        project_name = self.project.baseName()
+        if project_name:
+            pattern = re.compile("[\W_]+")  # NOQA
+            project_name = pattern.sub("", project_name)
+        else:
+            project_name = "CloudProject"
+        self.mProjectName.setText(project_name)
+        self.button_box.button(QDialogButtonBox.Save).setText(self.tr("Create"))
+        self.button_box.button(QDialogButtonBox.Save).clicked.connect(
+            self.convert_project
+        )
+
+        self.projectGroupBox.setVisible(True)
+        self.progressGroupBox.setVisible(False)
+
+        self.setup_gui()
+
+    def setup_gui(self):
+        """Populate gui and connect signals of the push dialog"""
+        project_file_name = QgsProject.instance().fileName()
+        if project_file_name:
+            export_folder_name = fileparts(project_file_name)[1]
+        else:
+            export_folder_name = "new_cloud_project"
+        export_folder_path = os.path.join(
+            self.qfield_preferences.value("cloudDirectory"), export_folder_name
+        )
+
+        self.mExportDir.setText(export_folder_path)
+        self.mExportDirBtn.clicked.connect(make_folder_selector(self.mExportDir))
+        self.update_info_visibility()
+
+    def get_export_folder_from_dialog(self):
+        """Get the export folder according to the inputs in the selected"""
+        return self.mExportDir.text()
+
+    def convert_project(self):
+        for cloud_project in self.network_manager.projects_cache.projects:
+            if cloud_project.name == self.mProjectName.text():
+                QMessageBox.warning(
+                    None,
+                    self.tr("Warning"),
+                    self.tr(
+                        "The project name is already present in your QFieldCloud repository, please pick a different name."
+                    ),
+                )
+                return
+
+        if sorted(Path(self.mExportDir.text()).rglob("*.qgs")) or sorted(
+            Path(self.mExportDir.text()).rglob("*.qgz")
+        ):
+            QMessageBox.warning(
+                None,
+                self.tr("Warning"),
+                self.tr(
+                    "The export directory already contains a project file, please pick a different directory."
+                ),
+            )
+            return
+
+        QApplication.setOverrideCursor(Qt.WaitCursor)
+
+        self.button_box.setEnabled(False)
+        self.projectGroupBox.setVisible(False)
+        self.progressGroupBox.setVisible(True)
+
+        cloud_convertor = CloudConverter(
+            self.project, self.get_export_folder_from_dialog()
+        )
+
+        cloud_convertor.warning.connect(self.show_warning)
+        cloud_convertor.total_progress_updated.connect(self.update_total)
+
+        try:
+            cloud_convertor.convert()
+        except Exception:
+            QApplication.restoreOverrideCursor()
```

Comment:
https://github.com/opengisch/QgisModelBaker/blob/master/QgisModelBaker/utils/qt_utils.py#L240-L249

Super nice utility class there. Except async, then it becomes more complicated :laughing: 

```py
with OverrideCursor(Qt.WaitCursor):
    do_something_and_restore_cursor_no_matter_what()
```